include/zephyr/bluetooth/audio/stream.h

```diff
@@ -0,0 +1,357 @@
+/** @file
+ *  @brief Bluetooth Audio Stream
+ */
```

Comment:
For CAP we need to add another layer on top of the `bt_audio_stream`, essentially making the `bt_audio_stream` into a `bt_bap_stream`, as the relationsship between BAP streams and CAP streams are similar to the relationsship between BAP and ISO, i.e. it is essentially an extension of the other. 

I had planned to rename `struct bt_audio_stream` into `struct bt_bap_stream` and move it into a `bap.h`. 

This can still be done after this change though, but it kind of defeats the purpose of this move. 

The structure I had in mind was a `bap.h` (containing BAP specific resources) and a `cap.h` containing CAP specific resources, and then keep an `audio.h` header for things that span across LE Audio in general (like the assigned numbers).